bench/hamming-cleanup/quadratic.fpcore

```diff
@@ -0,0 +1,127 @@
+; -*- mode: scheme -*-
+
+(FPCore (a b c)
+ :name "quadp (p42, positive)"
+ ;:herbie-expected #f ; Fails in single precision due to bad samples
+ :herbie-target
+ (let ([sqtD
+  (let ([x (* (sqrt (fabs a)) (sqrt (fabs c)))])
+    (if (or (and (> a 0) (> c 0)) (and (<= a 0) (<= c 0)))
+      (let ([acx
+        (if (and (> (fabs a) 1) (> x 1))
+                (* (/ (fabs a) x) (fabs c))
+                (* (fabs a) (/ (fabs c) x)))])
+      (let ([acxe 
+        (if (and (> (fabs a) 1) (> x 1))
+              (+ (fma (/ (fabs a) x) (- (fabs c)) acx) (* (/ (fma (/ (fabs a) x) x (- (fabs a))) x) (fabs c)))
+              (+ (fma (- (fabs a)) (/ (fabs c) x) acx) (* (/ (fma (/ (fabs c) x) x (- (fabs c))) x) a)))])
+      (let ([dp
+       (- (- (fabs b) x) (/ (- (- acx x) acxe) 2))])
+          (if (> dp 0) (* (sqrt dp) (sqrt (+ (fabs b) x)))
+          (if (== dp 0) 0
+            NAN)))))
+      (if (> (fabs b) x)
+        (* (fabs b) (sqrt (+ 1.0 (* (/ x (fabs b)) (/ x (fabs b))))))
+        (* x (sqrt (+ (* (/ (fabs b) x) (/ (fabs b) x)) 1.0))))))])
+  (if (== a 0) (- (/ c b))
+    (if (== sqtD NAN) NAN
+    (if (== sqtD 0) (- (/ (/ b 2) a))
+    (if (< b 0) (/ (- sqtD (/ b 2)) a)
+      (/ (- c) (+ (/ b 2) sqtD)))))))
```

Comment:
First of all, `(== sqtD NAN)` case is literally always false, `NAN` is not equal to anything, not even itself.

But more importantly I don't think we need any of these cases except the final `(if (< b 0)` case. The rest of these only apply in exceptional cases that Herbie will never sample.